mongodb/mongo_dataset.cc

```diff
@@ -0,0 +1,267 @@
+/**                                                                 -*- C++ -*-
+ * mongo_dataset.cc
+ * Mich, 2016-08-05
+ * This file is part of MLDB. Copyright 2015 Datacratic. All rights reserved.
+ **/
+#include <memory>
+#include <thread>
+#include <mutex>
+
+#include "bsoncxx/oid.hpp"
+#include "bsoncxx/builder/stream/document.hpp"
+#include "bsoncxx/builder/stream/array.hpp"
+#include "mongocxx/client.hpp"
+
+#include "mldb/core/function.h"
+#include "mldb/core/dataset.h"
+#include "mldb/types/structure_description.h"
+#include "mldb/rest/rest_request_router.h"
+#include "mldb/types/any_impl.h"
+
+#include "mongo_common.h"
+
+using namespace std;
+
+namespace Datacratic {
+namespace MLDB {
+namespace Mongo {
+
+typedef tuple<Path, CellValue, Date> Cell;
+
+struct MongoMatrixView : MatrixView {
+
+    vector<Path> getRowNames(ssize_t start = 0,
+                             ssize_t limit = -1) const override
+    {
+        return vector<Path>{};
```

Comment:
Can we remove MongoMatrixView and MongoColumnIndex and simply assert on getMatrixView and getColumnIndex?